### PR TITLE
Fixes broken server if snapshot file is faulty

### PR DESCRIPTION
**Pull request by ZeljkoBenovic**
# Description

If snapshots file is empty/corrupted for some reason, the server can't run, as return statements break code execution.

This PR provides better log output with problems with reading shapshots/metadata files.
Also, if there are any problems with these files, they will be deleted so that we can resync with the rest of the nodes.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

File: consensus/ibft/snapshot.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func (i *Ibft) setupSnapshot() error {
 
 	// Read from storage
 	if i.config.Path != "" {
-		if err := i.store.loadFromPath(i.config.Path); err != nil {
+		if err := i.store.loadFromPath(i); err != nil {
```
Comment (zivkovicmilos) [inline consensus/ibft/snapshot.go]: Please change the method to accept the path and the logger, no need to accept the entire `Ibft` reference

File: consensus/ibft/snapshot.go [hunk 2]
```diff
@@ -366,21 +366,29 @@ func newSnapshotStore() *snapshotStore {
...
+func (s *snapshotStore) loadFromPath(i *Ibft) error {
 	// Load metadata
 	var meta *snapshotMetadata
-	if err := readDataStore(filepath.Join(path, "metadata"), &meta); err != nil {
-		return err
-	}
+	if err := readDataStore(filepath.Join(i.config.Path, "metadata"), &meta); err != nil {
+		// if we can't read file metadata file delete it and sync again
+		// and logout that we've had some problems
+		i.logger.Error("could not read metadata file","err",err.Error())
```
Comment (zivkovicmilos) [inline consensus/ibft/snapshot.go]: Please change the log output from `metadata` to `snapshot store`, in the entire method

File: consensus/ibft/snapshot.go [hunk 3]
```diff
@@ -366,21 +366,29 @@ func newSnapshotStore() *snapshotStore {
...
+		i.logger.Error("removing faulty metadata file...")
```
Comment (zivkovicmilos) [inline consensus/ibft/snapshot.go]: This log should happen after you run `os.Remove` and it should go `Removed invalid snapshot store file`

File: consensus/ibft/snapshot.go (hunk 2 above)
Comment (zivkovicmilos) [reply consensus/ibft/snapshot.go]: Is there any chance for us to see the actual error and compare it before wiping out the snapshot store file?

**Review by zivkovicmilos — CHANGES_REQUESTED** [review]
Please resolve the linting errors and address pending comments 🙏 

Otherwise it's good to go 👍

File: consensus/ibft/snapshot.go (hunk 1 above)
Comment (ZeljkoBenovic) [reply consensus/ibft/snapshot.go]: Fixed with commit b0bf7cd004339996424fc7301b460a33873b367f

File: consensus/ibft/snapshot.go (hunk 2 above)
Comment (ZeljkoBenovic) [reply consensus/ibft/snapshot.go]: FIxed in commit b0bf7cd004339996424fc7301b460a33873b367f
The actual error will be outputted to console logs, something similar to this:
```
 2022-01-17T14:27:09.155Z [ERROR] polygon.consensus.ibft: Could not read snapshot data: err="unexpected end of JSON input"
```

File: consensus/ibft/snapshot.go (hunk 3 above)
Comment (ZeljkoBenovic) [reply consensus/ibft/snapshot.go]: Fixed with commit b0bf7cd004339996424fc7301b460a33873b367f

File: consensus/ibft/snapshot.go (hunk 2 above)
Comment (zivkovicmilos) [reply consensus/ibft/snapshot.go]: Got it, makes sense.
The file is empty so there's nothing to unmarshal for the json module

File: consensus/ibft/snapshot.go (hunk 1 above)
Comment (zivkovicmilos) [reply consensus/ibft/snapshot.go]: Please change out the `loadFromPath` method not to accept a spread, but instead a single logger instance

Comment (ZeljkoBenovic) [reply consensus/ibft/snapshot.go]: Fixed with commit 576cbe70a70e335a395994055be9715d8c609c27

**Review by Kourin1996 — APPROVED** [review]
Looks good to me

**Review by zivkovicmilos — APPROVED** [review]
Lookin' good 🚀

**Pull request merged**